### PR TITLE
Drivers: Using retryAfterSeconds in more places in drivers vs. retryAfterMs

### PR DESCRIPTION
**Pull request by vladsud**
It seems like we are converging on using seconds across various driver layers, not ms.
But there are couple places where we convert back and forth.
Trying to fix it by using seconds in more workflows.

Please let me know if it feels better or worse :)

### CODE REVIEW

**Comment by curtisman** [comment]
I think we were trying to converge to millisecond instead?

https://github.com/microsoft/FluidFramework/pull/6261
https://github.com/microsoft/FluidFramework/pull/6030

**Comment by vladsud** [comment]
Second PR makes sense - layers that consume error and do waits use milliseconds.
They all should use getRetryDelayFromError() that calculates milliseconds and not directly care how data is encoded in error objects.

But first part - not sure. The reason I find it really weird is we do conversions back and forward, which increases chances of errors:

```typescript
const retryAfterMs = retryAfterSeconds !== undefined ? retryAfterSeconds * 1000 : undefined;
error = createGenericNetworkError(errorMessage, true, retryAfterMs, { statusCode });

export function createGenericNetworkError(
    errorMessage: string,
    canRetry: boolean,
    retryAfterMs?: number,
    props?: ITaggableTelemetryProperties) {
    if (retryAfterMs !== undefined && canRetry) {
        return new ThrottlingError(errorMessage, retryAfterMs / 1000, props);
    }
    return new GenericNetworkError(errorMessage, canRetry, props);
```

I think we either should keep these layers using seconds all the way (as almost all errors in protocol are seconds, and I think it would be great for IR11sSocketError to also use seconds as that's the only exception, or fork SPO / R11 helper functionality as SPO always uses seconds on all paths.)

OR we should change underlying error object structure (need to consider going through compat layer) and error objects to use milliseconds, not seconds.

But converting 2 times back and forth seems weird.

File: packages/drivers/routerlicious-driver/src/errorUtils.ts [hunk 1]
```diff
@@ -96,6 +96,6 @@ export function errorObjectFromSocketError(socketError: IR11sSocketError, handle
     return createR11sNetworkError(
         message,
         socketError.code,
-        socketError.retryAfterMs,
+        socketError.retryAfterMs === undefined ? undefined : socketError.retryAfterMs * 1000,
```
Comment (chensixx) [inline packages/drivers/routerlicious-driver/src/errorUtils.ts:99]: should this be called socketError.retryAfterSeconds?

File: packages/loader/driver-utils/src/network.ts [hunk 2]
```diff
@@ -129,11 +129,11 @@ export const createWriteError = (errorMessage: string) =>
...
     errorMessage: string,
     canRetry: boolean,
-    retryAfterMs?: number,
+    retryAfterSeconds?: number,
     props?: ITelemetryProperties,
 ): ThrottlingError | GenericNetworkError {
-    if (retryAfterMs !== undefined && canRetry) {
-        return new ThrottlingError(errorMessage, retryAfterMs / 1000, props);
+    if (retryAfterSeconds !== undefined && canRetry) {
+        return new ThrottlingError(errorMessage, retryAfterSeconds, props);
```
Comment (andre4i) [inline packages/loader/driver-utils/src/network.ts:136]: It feels better as long as we don't care about this granularity. But if we only care about consistency, probably changing ThrottlingError to use milliseconds would make the change have a smaller impact.

Comment (vladsud) [reply packages/loader/driver-utils/src/network.ts:136]: We can, but such change has implications (i.e. getRetryDelayFromError and impact on back compat between layers).
Plus all telemetry would be in milliseconds (as it gets logged).
It's all doable, question is - what we want :)

File: packages/drivers/routerlicious-driver/src/errorUtils.ts (hunk 1 above)
Comment (vladsud) [reply packages/drivers/routerlicious-driver/src/errorUtils.ts:99]: That's part of protocol, so change like that is possible, but needs to be staged.
That's what I alluded in my comment above - ODSP uses time in seconds in protocol, it's not required but would be great for R11s to be the same here

**Comment by tylerbutler** [comment]
Naive question: do the functions accept floats as input or does it require ints?

File: packages/loader/driver-utils/src/network.ts (hunk 2 above)
Comment (andre4i) [reply packages/loader/driver-utils/src/network.ts:136]: Honestly I'd use something like a `Duration` class and make this less opinionated.

**Pull request closed without merging**